src/Microsoft.AspNet.Razor/Parser/CSharpCodeParser.cs

```diff
@@ -316,12 +316,15 @@ private void VerbatimBlock()
                 Span.ChunkGenerator = SpanChunkGenerator.Null;
             }
 
-            if (!At(CSharpSymbolType.WhiteSpace) && !At(CSharpSymbolType.NewLine))
+            if (IsNested)
             {
-                PutCurrentBack();
+                CompleteBlock(insertMarkerIfNecessary: false);
```

Comment:
Hmm. The name is misleading here. This `CompleteBlock` does nothing here other than `EnsureCurrent` and `PutCurrentBack`. Removing it affects no tests. I am thinking we should remove it completely.